src/utils/daos_hdlr.c

```diff
@@ -3006,7 +3008,8 @@ dm_parse_path(struct file_dfs *file, char *path, size_t path_len,
 		file->type = POSIX;
 	}
 out:
-	D_FREE(dattr.da_rel_path);
+	if (dattr.da_rel_path)
+		free(dattr.da_rel_path);
```

Comment:
Curious why this instead of D_FREE?

Reply:
because this can be used and freed externally too (outside DAOS). and so we need to allocate it without the D_ macros. otherwise memory tracing / logging will fail

Reply:
I hadn't considered that before. Thank you

Reply:
There's a D_REALPATH() already

Reply:
ah. then i will change to use that

Reply:
note that my comment applies to the other free of realp, not this one.. this still needs to be a free because the free can happen externally to the library